src/librustc/diagnostics.rs

```diff
@@ -984,6 +984,57 @@ From [RFC 246]:
 [RFC 246]: https://github.com/rust-lang/rfcs/pull/246
 "##,
 
+E0395: r##"
+The value assigned to a constant expression must be known at compile time,
+which is not the case when comparing raw pointers. Erroneous code example:
+
+```
+static foo: i32 = 42;
+static bar: i32 = 43;
+
+static baz: bool = { (&foo as *const i32) == (&bar as *const i32) };
+// error: raw pointers cannot be compared in statics!
+```
+
+To fix this error, please do not assign this value to a constant
```

Comment:
This is not the fix.

Avoid recommending catchall fixes like this, since they're not applicable to all situations (but the language makes them sound like it)

Mention common causes and suggest fixes for those instead.